dockerspawner/dockerspawner.py

```diff
@@ -616,6 +617,10 @@ def get_ip_and_port(self):
                 raise RuntimeError("Failed to get port info for %s" % self.container_id)
             ip = resp[0]['HostIp']
             port = int(resp[0]['HostPort'])
+
+        if ip == '0.0.0.0':
+            ip = os.environ['DOCKER_HOST'].split(':')[0]
```

Comment:
Indeed - it looks like the docker host could also be passed in from `self.client_kwargs`:
https://github.com/jupyterhub/dockerspawner/blob/a272e165b79158a14ee973ec505a1d4dbc3acf28/dockerspawner/dockerspawner.py#L60-L62

In which case it might be better to pull it off the `APIClient` directly. Will look at doing that...